### ISSUE TITLE
Footnote support

### ISSUE THREAD

**Issue by EiffelFly** [kentcdodds/mdx-bundler#73]
Hi, thank you for your hard working, really love the flexibility of mdx-bundler.

After searching for a while and testing, it seems that mdx-bundler doesn't support markdown footnote syntax,

```markdown
This paragraph has a footnote.[^1]
[^1]: Here's the footnote
```

should be rendered like

```html
<p>
  This paragraph has a footnote.<sup><a href="#fn1" id="fnref1">1</a></sup>
</p>

Here's the footnote. <a href="#fnref1">↩</a>
```

Really look forward to this feature.

Thank you!!

**Comment by Arcath** [comment]
Hmmm, this seems like it might be an XDM issue. Paging @wooorm

**Comment by wooorm** [comment]
footnotes are not a markdown feature. Try them on GitHub or Reddit and they won’t work. They are a custom syntax extension without a spec. As is often the case with custom things, there are plugins for those: https://github.com/remarkjs/remark-footnotes.

**Comment by Arcath** [comment]
@wooorm Thanks :)

@EiffelFly mdx-bundler supports remark plugins in its [`xdmOptions` parameter](https://github.com/kentcdodds/mdx-bundler#xdmoptions) so this should be pretty easy to implement.

**Comment by EiffelFly** [comment]
You guys are awesome!!! 

@wooorm thanks for the tips. 

@Arcath Thank for your kindly help, I have also read your blog too, very helpful.

**Comment by EiffelFly** [comment]
@Arcath I implement this features today but without luck, here is my prepareMDX

```js
import { bundleMDX } from "mdx-bundler";
import path from "path";
import remarkFootnotes from "remark-footnotes";

export const prepareMDX = async (source, files, cwd) => {
  process.env.ESBUILD_BINARY_PATH = path.join(
    process.cwd(),
    "node_modules",
    "esbuild",
    "bin",
    "esbuild"
  );

  const { code } = await bundleMDX(source, {
    cwd: cwd,
    files: files,
    xdmOptions: (options) => {
      options.remarkPlugins = [
        ...(options.remarkPlugins ?? []),
        remarkFootnotes,
      ];
      return options;
    },
  });

  return code;
};

```

and the way I get the content

```js
export async function getContent(slug, locale, type = "post") {
  const dir = type === "page" ? pagesDirectory : postsDirectory;
  const fullPath = path.join(dir, slug, `index.${locale}.mdx`);
  const fileContents = fs.readFileSync(fullPath, "utf-8");
  const { content, data } = matter(fileContents.trim());
  let components = {};

  // if ( data.components.length > 0 ){
  //   components = getComponents(data.components, dir, slug);
  // }

  const code = await prepareMDX(content, components, path.join(dir, slug));
  return {
    slug: slug,
    code: code,
    metadata: data,
  };
}
```

my testing MDX file

```md

[^1] hi, testing

- [^1] yo
```


There is no error message, but footnotes won't show. Could you please light up some path for me.

**Comment by EiffelFly** [comment]
I have figured it out!!!

remark-footnote v2.0.0 have some issues are fixed at v3.0.0.

Thankfully xdm is using remark-prase @9.0.0 so we can use remark-footnote v3.0.0 and everything work smoothly!

**Comment by Arcath** [comment]
Awesome :) glad you got iy working!
